src/client/play.js

```diff
@@ -1,25 +1,43 @@
 const states = require('../states')
-const crypto = require('crypto')
-const concat = require('../transforms/binaryStream').concat
+const signedChatPlugin = require('./chat')
 
 module.exports = function (client, options) {
+  client.serverFeatures = {}
+  client.on('server_data', (packet) => {
+    client.serverFeatures = {
+      chatPreview: packet.previewsChat,
+      enforcesSecureChat: packet.enforcesSecureChat
+    }
+  })
+
   client.once('success', onLogin)
 
   function onLogin (packet) {
+    const mcData = require('minecraft-data')(client.version)
     client.state = states.PLAY
     client.uuid = packet.uuid
     client.username = packet.username
-    client.signMessage = (message, timestamp, salt = 0) => {
-      if (!client.profileKeys) throw Error("Can't sign message without profile keys, please set valid auth mode")
-      const signable = concat('i64', salt, 'UUID', client.uuid, 'i64',
-        timestamp / 1000n, 'pstring', JSON.stringify({ text: message }))
-      return crypto.sign('RSA-SHA256', signable, client.profileKeys.private)
+
+    if (mcData.supportFeature('signedChat')) {
+      if (!options.disableChatSigning && client.serverFeatures.enforcesSecureChat) {
```

Comment:
Shouldn't this be `options.disableChatSigning && client.serverFeatures.enforcesSecureChat` instead?